tests/test_tensornet.py

```diff
@@ -0,0 +1,570 @@
+
+        @qml.qnode(tensornet_device_1_wire)
+        def circuit():
+            op(wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        assert np.isclose(circuit(), expected_output, atol=tol, rtol=0)
+
+    # This test is ran against the state |Phi+> with two Z expvals
+    @pytest.mark.parametrize("name,expected_output", [
+        ("CNOT", [-1/2, 1]),
+        ("SWAP", [-1/2, -1/2]),
+        ("CZ", [-1/2, -1/2]),
+    ])
+    def _test_supported_gate_two_wires_no_parameters(self, tensornet_device_2_wires, tol, name, expected_output):  # TODO: change name when QubitStateVector is supported
+        """Tests supported gates that act on two wires that are not parameterized"""
+
+        op = getattr(qml.ops, name)
+
+        assert tensornet_device_2_wires.supports_operation(name)
+
+        @qml.qnode(tensornet_device_2_wires)
+        def circuit():
+            qml.QubitStateVector(np.array([1/2, 0, 0, math.sqrt(3)/2]), wires=[0, 1])
+            op(wires=[0, 1])
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))
+
+        assert np.allclose(circuit(), expected_output, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("name,expected_output", [
+        ("CSWAP", [-1, -1, 1]),
+    ])
+    def _test_supported_gate_three_wires_no_parameters(self, tensornet_device_3_wires, tol, name, expected_output):  # TODO: change name when BasisState is supported
+        """Tests supported gates that act on three wires that are not parameterized"""
+
+        op = getattr(qml.ops, name)
+
+        assert tensornet_device_3_wires.supports_operation(name)
+
+        @qml.qnode(tensornet_device_3_wires)
+        def circuit():
+            qml.BasisState(np.array([1, 0, 1]), wires=[0, 1, 2])
+            op(wires=[0, 1, 2])
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1)), qml.expval(qml.PauliZ(2))
+
+        assert np.allclose(circuit(), expected_output, atol=tol, rtol=0)
+
+    # This test is ran with two Z expvals
+    @pytest.mark.parametrize("name,par,expected_output", [
+        ("BasisState", [0, 0], [1, 1]),
+        ("BasisState", [1, 0], [-1, 1]),
+        ("BasisState", [0, 1], [1, -1]),
+        ("QubitStateVector", [1, 0, 0, 0], [1, 1]),
+        ("QubitStateVector", [0, 0, 1, 0], [-1, 1]),
+        ("QubitStateVector", [0, 1, 0, 0], [1, -1]),
+    ])
+    def _test_supported_state_preparation(self, tensornet_device_2_wires, tol, name, par, expected_output):   # TODO: change name when QubitStateVector and BasisState are supported
+        """Tests supported state preparations"""
+
+        op = getattr(qml.ops, name)
+
+        assert tensornet_device_2_wires.supports_operation(name)
+
+        @qml.qnode(tensornet_device_2_wires)
+        def circuit():
+            op(np.array(par), wires=[0, 1])
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))
+
+        assert np.allclose(circuit(), expected_output, atol=tol, rtol=0)
```

Comment:
Not major but I would argue we should have separate tests for BasisState and QubitStateVector.

Reply:
parametrizations should be used for test data, not objects with potentially different functionality/logic flows in the device

Reply:
(I assume default qubit is like this)

Reply:
Yes, copied directly from `test_default_qubit.py`.
To be honest, I think we should just have a single `test_qubit_simulators.py`, which uses `parameterize` to test both devices

Reply:
That might make sense for an integration test (e.g., testing applications of gates and measuring observables), but we should still have separate unit tests per plugin to test specific logic and warnings.

Reply:
Yep that's what I was thinking

Reply:
The integration tests in the [XanaduAI/pennylane-plugin-template](https://github.com/XanaduAI/pennylane-plugin-template) repository were based on default qubit but written to be device independent, we could just drop them in. There are separate `test_apply.py`, `test_expval`, `test_sample`, and `test_var` integration tests there.